scrapers/nd/events.py

```diff
@@ -1,114 +1,174 @@
-import dateutil.parser
-import lxml
-import pytz
 import re
+import pytz
+import logging
+import dateutil.parser
+import datetime
+import requests
+import lxml.html
+from spatula import HtmlPage
 from openstates.scrape import Scraper, Event
 
 
-class NDEventScraper(Scraper):
-    _tz = pytz.timezone("US/Central")
-    event_months = set()
-    events = {}
-
-    def scrape(self, session=None):
-
-        if not session:
-            session = self.latest_session()
-            self.info("no session specified, using %s", session)
-        # figuring out starting year from metadata
-        for item in self.jurisdiction.legislative_sessions:
-            if item["identifier"] == session:
-                start_year = item["start_date"][:4]
-                self.year = start_year
-                break
-
-        url = f"https://www.legis.nd.gov/assembly/{session}-{start_year}/committees/interim/committee-meeting-summary"
-
-        page = self.get(url).content
-        page = lxml.html.fromstring(page)
-        page.make_links_absolute(url)
-
-        for table in page.xpath('//table[contains(@class,"views-table")]'):
-            com = table.xpath("caption/a")[0].text_content().strip()
-            for row in table.xpath("tbody/tr"):
-                date_link = row.xpath("td[1]/strong/a")[0]
-                event_url = date_link.xpath("@href")[0]
-
-                date = date_link.xpath("span")[0].text_content().strip()
-                date = dateutil.parser.parse(date)
-                date = self._tz.localize(date)
-
-                self.event_months.add(date.strftime("%Y-%m"))
-
-                location = "See Agenda"
-
-                event = Event(name=com, start_date=date, location_name=location)
-
-                event.add_source(event_url)
-
-                for link in row.xpath("td[2]//a"):
-                    link_text = link.text_content().strip()
-
-                    # skip live broadcast links
-                    if "video.legis" in link_text:
-                        continue
+def time_is_earlier(new, current):
+    new_magnitude = int(new[0:2] + new[3:5])
+    current_magnitude = int(current[0:2] + current[3:5])
+    if new_magnitude < current_magnitude:
+        return True
+    else:
+        return False
+
+
+class EventConsolidator:
+    def __init__(self, items, url):
+        self.items = items
+        self.events = {}
+        self._tz = pytz.timezone("US/Central")
+        self.url = url
+
+    def consolidate(self):
+        for item in self.items:
+            date_time = item["date_time"]
+            local_date = dateutil.parser.parse(date_time)
+
+            item_date, item_time = str(local_date).split()
+
+            committee = item["committee"]
+            location = item["location"]
+            event_key = f"{item_date}-{committee}-{location}"
+
+            if not self.events.get(event_key):
+                self.events[event_key] = {}
```

Comment:
```
self.events[event_key] = {
    "event_start_time": item_time,
    "item_keys": set(),
}
```
Is probably cleaner here.

Reply:
Definitely cleaner. I [changed the construction](https://github.com/openstates/openstates-scrapers/pull/4162/commits/037209511528934a8e11f7616c83939e636b489a#diff-fa16689e28573ece469e8dfea222521c243ef203a1c8c974d48d6b050b3ed308R40-R43) accordingly.